README.md

```diff
@@ -89,6 +89,33 @@ MyClass instance = ServiceLocator.locate().resolve(MyClass.class);
 ServiceLocator.reset(); // Clear all registrations
 ```
 
+### Unit of Work 
+Implementation of a pattern-agnostic (trying to be) Unit of Work pattern in Java. For now the only implementation is using JDBI but it could be easily adapted to pure JDBC or Hibernate.
+
+**Warning** 
+> The JDBI UnitOfWork uses `ThreadLocal` for its "context". You will not get transaction-isolation in "sub-threads" or asynchronous code.
+
+```java
+UnitOfWork<Handle> unitOfWork = new JDBIUnitOfWork(DBI);
+unitOfWork.begin();
+
+try {
+	// never use simple double for currency manipulations, this is an example
+	unitOfWork.withTransaction(transaction -> withdraw("user1", 5.0));
+	unitOfWork.withTransaction(transaction -> withdraw("user2", 5.0));
+	unitOfWork.withTransaction(transaction -> transfer("user3", 10.0));
+
+	unitOfWork.commit();
+} catch (CriticalMoneyException e) {
```

Comment:
Even though the example is pretty good, I think I would not use a money-related example and simplify the example. It is not complex, but I feel I need to argue with the double topic 🤓